app/src/main/java/com/arctouch/publictransportation/activity/RouteListActivity.java

```diff
@@ -0,0 +1,193 @@
+package com.arctouch.publictransportation.activity;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.EditorInfo;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.android.volley.RequestQueue;
+import com.arctouch.publictransportation.R;
+import com.arctouch.publictransportation.domain.Route;
+import com.arctouch.publictransportation.domain.generic.params.StopNameParam;
+import com.arctouch.publictransportation.domain.generic.returns.RoutesReturn;
+import com.arctouch.publictransportation.fragment.DepartureListFragment;
+import com.arctouch.publictransportation.fragment.StopListFragment;
+import com.arctouch.publictransportation.interfaces.VolleyListener;
+import com.arctouch.publictransportation.volley.VolleySingleton;
+import com.arctouch.publictransportation.volley.endpoints.RoutesVolley;
+
+import java.util.List;
+
+public class RouteListActivity extends AppCompatActivity implements VolleyListener<RoutesReturn> {
+    private View mRecyclerView;
+    private ProgressBar mProgressBar;
+    private TextView mEmptyList;
+
+    private boolean mMultiplePane;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_route_list);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        toolbar.setTitle(getTitle());
+
+        EditText mTextSearch = (EditText) findViewById(R.id.textSearch);
+        mTextSearch.setOnEditorActionListener(searchAction);
+
+        this.mProgressBar = (ProgressBar) findViewById(R.id.progressBar);
+        this.mEmptyList = (TextView) findViewById(R.id.emptyList);
+
+        this.mRecyclerView = findViewById(R.id.routeList);
+        assert mRecyclerView != null;
+
+        // Only show in large-screen layouts
+        if (findViewById(R.id.stopListContainer) != null) {
+            mMultiplePane = true;
+        }
+    }
+
+    private TextView.OnEditorActionListener searchAction = new TextView.OnEditorActionListener() {
+        @Override
+        public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+            if (actionId == EditorInfo.IME_ACTION_SEARCH) {
```

Comment:
You don't actually have to restrict this to _IME_ACTION_SEARCH_; You've already set _actionSearch_ on the textView, so the keyboard can only send that one action.
(Also, there's always the possibility a user has a cheap phone with a custom Android version of an old api that doesn't behave well with imeOptions...)